airflow/providers/cncf/kubernetes/utils/pod_manager.py

```diff
@@ -614,15 +622,18 @@ def await_pod_completion(
         :param container_name: name of the container within the pod
         :return: tuple[State, str | None]
         """
-        while True:
-            remote_pod = self.read_pod(pod)
-            if remote_pod.status.phase in PodPhase.terminal_states:
-                break
-            if istio_enabled and container_is_completed(remote_pod, container_name):
-                break
-            self.log.info("Pod %s has phase %s", pod.metadata.name, remote_pod.status.phase)
-            time.sleep(2)
-        return remote_pod
+        remote_pod = self._client.read_namespaced_pod_status(pod.metadata.name, pod.metadata.namespace)
+        self.log.info("Pod %s has phase %s", pod.metadata.name, remote_pod.status.phase)
+
+        if any(
+            [
+                remote_pod.status.phase in PodPhase.terminal_states,
+                istio_enabled and container_is_completed(remote_pod, container_name),
+            ]
```

Comment:
```suggestion
            (
                remote_pod.status.phase in PodPhase.terminal_states,
                istio_enabled and container_is_completed(remote_pod, container_name),
            )
```
i think this avoids making a list object

Reply:
Is there any real benefit to using tuple[bool] instead of a list[bool]?

Reply:
crap i was thinking it would be generator not tuple :)

Reply:
as in `(x for x in some thing)` vs `[x for x in some thing]`

but that's not the case here